crypto/bls/src/secret_key.rs

```diff
@@ -42,52 +43,21 @@ impl SecretKey {
     }
 
     /// Returns the underlying secret key.
-    pub fn as_raw(&self) -> &RawSecretKey {
+    pub(crate) fn as_raw(&self) -> &RawSecretKey {
         &self.0
     }
 }
 
-impl_ssz!(SecretKey, BLS_SECRET_KEY_BYTE_SIZE, "SecretKey");
-
-impl_tree_hash!(SecretKey, BLS_SECRET_KEY_BYTE_SIZE);
-
-impl Serialize for SecretKey {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        serializer.serialize_str(&hex_encode(ssz_encode(self)))
-    }
-}
-
-impl<'de> Deserialize<'de> for SecretKey {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let bytes = deserializer.deserialize_str(PrefixedHexVisitor)?;
-        let secret_key = SecretKey::from_ssz_bytes(&bytes[..])
-            .map_err(|e| serde::de::Error::custom(format!("invalid ssz ({:?})", e)))?;
-        Ok(secret_key)
-    }
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
-    use ssz::ssz_encode;
 
     #[test]
-    pub fn test_ssz_round_trip() {
+    pub fn from_bytes() {
         let byte_key = [
             3, 211, 210, 129, 231, 69, 162, 234, 16, 15, 244, 214, 126, 201, 0, 85, 28, 239, 82,
             121, 208, 190, 223, 6, 169, 202, 86, 236, 197, 218, 3, 69,
         ];
-        let original = SecretKey::from_bytes(&byte_key).unwrap();
-
-        let bytes = ssz_encode(&original);
-        let decoded = SecretKey::from_ssz_bytes(&bytes).unwrap();
-
-        assert_eq!(original, decoded);
+        SecretKey::from_bytes(&byte_key).unwrap();
```

Comment:
Should we test the round trip?
```rust
        let key = SecretKey::from_bytes(&byte_key).unwrap();
        assert_eq!(byte_key, key.as_bytes().as_bytes());
```

Reply:
Yep good call, deleted it in an (unsuccessful) attempt to fully remove `.as_bytes()`, forgot to add it back.